### ISSUE TITLE
Can't execute script on remote by piping through ssh

### ISSUE THREAD

**Issue by elbisbe** [microsoft/vscode-remote-release#1899]
Hello!

I've been trying for a while to set up the Remote-SSH extension at my workplace to no avail.
I have to connect using keyboard-interactive auth, that launches into a corporative login banner. I cannot use public-key authentication for the time being.
The problem I'm finding here is that I can connect to the server, but as soon as I authenticate it hangs for some seconds before closing the connection.

I've tried to ssh on PowerShell and I was able to connect properly as long as i didn't put a command in the ssh, that I imagine is getting in conflict with the login banner. I'm seeing that by default on the remote ssh the connection commands put a "bash" as a command in the ssh.

Is there any way to disable this behavior? I guess the bash command is needed for further install of the vscode server, but I've run out of ideas on how to tackle this problem.

I attach the output of the Remote-SSH, in case its useful:
```
[10:53:24.672] Running script with connection command: ssh -T -o ClearAllForwardings=true host bash
[10:53:24.676] Install and start server if needed
[10:53:24.686] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[10:53:25.065] > 
[10:53:25.065] Got some output, clearing connection timeout
[10:53:25.250] > Password:
[10:53:30.593] > 
> 
[10:53:36.989] "install" terminal command done
[10:53:36.989] Install terminal quit with output: Password:
[10:53:36.989] Received install output: Password:
[10:53:36.991] Stopped parsing output early. Remaining text: Password:
[10:53:36.991] Failed to parse remote port from server output
[10:53:36.991] Resolver error: 
[10:53:36.995] TELEMETRY: {"eventName":"resolver","properties":{"outcome":"failure","reason":"UnparsableOutput","askedPw":"0","askedPassphrase":"0","asked2fa":"0","askedHostKey":"0","gotUnrecognizedPrompt":"0","remoteInConfigFile":"1"},"measures":{"resolveAttempts":1,"retries":1}}
[10:53:36.996] ------
```
So "ssh user@host" works, but "ssh user@host bash" doesn't.

Any suggestion would be appreciated. Sorry if this has been solved before, I've haven't been able to find it anywhere.

Cheers!

**Comment by roblourens** [comment]
The extension needs to be able to run one command to connect and run a command on the remote. I don't know what your setup is like but if you can find a way to do this with one command, I can probably help you out?

**Comment by elbisbe** [comment]
It unfortunately seems that it's not allowed to pass commands on the connection as a security measure, just only after you auth in the system. My initial question asked whether I could modify the 

> ssh -T -o ClearAllForwardings=true host bash

command in some way. If that cannot be done, I'm done with the issue as it's a problem on my end. Thanks for the support!

**Comment by roblourens** [comment]
We are actually doing something like `echo longcommand | ssh host bash`. If there's some way you could modify that command to work in your system, then I'd like to add the ability to do that.

**Comment by elbisbe** [comment]
Hey! Sorry for the delay, we've been having some national holidays and haven't been able to try out some ideas. For security measures I have a custom login banner script configured through the sshd that takes any command I pass through ssh, so I need to deal with the login banner first and then I can start using commands.

A posible solution for me would be if I had the ability to do the `longcommand` on my own, or at least have any way to delay the execution of `longcommand` until I tell him to do so. Having a way to manually run the `longcommand` would probably solve my issue. I understand that it would break the automatic config and it would mean that I should do it every time I connect to the SSH, but it would be the most I can hope given my situation. 

TL;DR: I can't pass commands through ssh, but I could run a script that does the `longcommand` so I could run it.

**Comment by roblourens** [comment]
Ok thanks for the details. I think there are other cases that would be served by having the user just set things up manually, but I'm not exactly sure how that would look.

**Comment by ronaknnathani** [comment]
FWIW, same problem on my end too. Our workstations have a custom login banner and `echo <command> | ssh user@host bash` doesn't display the output of the command piped through ssh.

**Comment by tanhakabir** [comment]
Not all scripts will work but some it's possible that some scripts will work for you all if you use RemoteCommand which was recently enabled in `v0.72`. You can read more about it here: https://github.com/microsoft/vscode-docs/blob/main/remote-release-notes/v1_64.md#enable-remotecommand

**Comment by smartwaystudio** [comment]
Same problem here, we have a custom login banner and I'm not able to connect to server because of this.

**Comment by VSCodeTriageBot** [comment]
This issue is being closed to keep the number of issues in our inbox on a manageable level, we are closing issues that are not going to be addressed in the foreseeable future: We look at the number of votes the issue has received and the number of duplicate issues filed. More details [here](https://aka.ms/vscode-out-of-scope). If you disagree and feel that this issue is crucial: We are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Thanks for your understanding and happy coding!

**Comment by roblourens** [comment]
This is just something that we will have to require, but you might check out https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-server
